### DISCUSSION TITLE
Idea: allow LFO-functions in EEL formula fields

### DISCUSSION THREAD

**Discussion by vonglan** [helgoboss/helgobox#355] [Ideas]
Might lead to some nice use cases.
Allow a function `lfo_value(<frequency>,<shape>)` in EEL formula fields.
The result depends on the current play position and the parameters:
`<frequency>` is based on the length of the current measure, e.g. 0.25 for quarter beats
`<shape>` is 1-4 (sine, square, saw, triangle)
Output of the function is a value between -1 and +1.

**Comment by helgoboss** [comment]
Doesn't this kind of assume that you have a continuous stream of control messages?

**Comment by vonglan** [comment]
What I had in mind was a continuously running LFO - but you are right, that would only work with continuous control messages, or it would be a completely different concept from what ReaLearn is doing at the moment.
Maybe not a good idea.

**Reply by helgoboss** [reply]
Another perspective on it that maybe makes sense: Add "Project position" as source and and use sin, cos and so on. That would be very composable.

**Comment by vonglan** [comment]
I think the crucial questions are
- Is there a use case for this (that isn't already conveniently covered by existing JSFX)
- Does the use case justify the completely different concept (not a normal mapping)

I am not sure about either of those. Just had the idea. If it doesn't resonate with anyone, forget it.

**Comment by helgoboss** [comment]
Mmh true, one could just use a JSFX to generate some 14-bit LFO-like MIDI messages and feed them into ReaLearn.
